perfeed/llms/ollama_client.py

```diff
@@ -7,6 +7,7 @@
 
 class OllamaClient(BaseClient):
     def __init__(self, model: str):
+        self.provider = 'ollama'
```

Comment:
Why do we need this and where is it used?

Reply:
I see, but it feels to be that `self.provider` is an implicit field that can be kinda hidden to the user. An explicit way to do so is to create an 
```
@abstract
def get_provider(self):
```
on the `BaseClient` so it forces all implementation to explicitly specify the `provider`.

Reply:
gonna use class().__class__.name__